internal/pkg/agent/cmd/run.go

```diff
@@ -109,11 +109,15 @@ func run(override cfgOverrider, modifiers ...component.PlatformModifier) error {
 	if cfg.Settings.LoggingConfig != nil {
 		logLvl = cfg.Settings.LoggingConfig.Level
 	}
-	l, err := logger.NewFromConfig("", cfg.Settings.LoggingConfig, true)
+	baseLogger, err := logger.NewFromConfig("", cfg.Settings.LoggingConfig, true)
 	if err != nil {
 		return err
 	}
 
+	l := baseLogger.With("component", map[string]interface{}{
```

Comment:
I don't necessarily like the idea of `component.binary`. This conflates the idea of a component. Too Elastic Agent a component is something that it spawns and runs. By placing `component.binary` it makes it feel like that the Elastic Agent also spawns another Elastic Agent.

I still think the idea that `component` is missing in the logs means it's the Elastic Agent. Simple `grep -E` would show only the Elastic Agent logs.

For debugging and input development I think the idea of a `elastic-agent logs` subcommand that allows filter locally in the console would be very helpful and the `grep -E` could be done in that code. There is no need to actually add a field.

If we do want to add a field, I don't think `component.binary` or `service.name` are proper names for these logs.

Reply:
Is there a name you do like for this to avoid more back and forth if Michal tries this again? I think `component.binary` is better than `service.name`, and `component.binary` is already in the Fleet mappings for logs. Fleet is using the destination dataset for filtering so maybe including the `dataset` like we already do for components is a more sensible to to do this?

I don't mind the idea of a tool to grab the right the logs (or to analyze diagnostics easily in general), but that is another thing we'd have to write and maintain.

Reply:
An inverted match on "component:{` or similar will also exclude logs generated by the agent about components, but not from components. For example it would exclude:

```json
{"log.level":"info","@timestamp":"2023-01-10T18:21:35.428Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":825},"message":"Component state changed http/metrics-monitoring (FAILED->STARTING): Starting: spawned pid '42223'","component":{"id":"http/metrics-monitoring","state":"STARTING","old_state":"FAILED"},"ecs.version":"1.6.0"}
```

Reply:
cc @blakerouse because i dont' have anything against having component name for logging purposes

Reply:
Actually I think @cmacknz point above made me think of another issue with `component.*` fields. Being that Elastic Agent adds those fields in some of the logs that the Elastic Agent produces and always for the logs produced by the components, that means that filtering on `component.*` would show both logs from Elastic Agent and logs from the component. I don't think that is the behavior we want.

I think it would be best to provide a field that would allow filter to be done on wether the log comes from Elastic Agent or not.

So maybe we should add something like `log.source`.

Possible values would be:

- `log.source: elastic-agent` - Comes from the Elastic Agent
- `log.source: ${component-id}` - Comes from a specific component